app/services/cert/certs.rb

```diff
@@ -12,20 +12,28 @@ def self.candlepin_client_ca_cert
       File.read(SETTINGS[:katello][:candlepin][:ca_cert_file])
     end
 
-    def self.ssl_client_cert
-      @ssl_client_cert ||= OpenSSL::X509::Certificate.new(File.read(ssl_client_cert_filename))
+    def self.ssl_client_cert(use_admin_as_cn_cert: false)
+      @ssl_client_cert ||= OpenSSL::X509::Certificate.new(File.read(ssl_client_cert_filename(use_admin_as_cn_cert: use_admin_as_cn_cert)))
     end
 
-    def self.ssl_client_cert_filename
-      Setting['pulp_client_cert']
+    def self.ssl_client_cert_filename(use_admin_as_cn_cert: false)
+      if use_admin_as_cn_cert
+        '/etc/pki/katello/certs/pulp-client.crt'
+      else
+        Setting['ssl_certificate']
```

Comment:
In Foreman itself we always use symbols
```suggestion
        Setting[:ssl_certificate]
```